plugins/dell/fu-plugin-dell-esrt.c

```diff
@@ -0,0 +1,356 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2017 Dell, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <appstream-glib.h>
+#include <smbios_c/token.h>
+#include <smbios_c/smi.h>
+
+#include "fu-plugin.h"
+#include "fu-plugin-vfuncs.h"
+
+#include "dell-wmi-smi.h"
+
+static gboolean
+fu_plugin_dell_esrt_wmi_supported (void)
+{
+	if (access (DELL_WMI_CHAR, F_OK) != -1)
+		return TRUE;
+	return FALSE;
+}
+
+static gboolean
+fu_plugin_dell_esrt_wmi_call_ioctl (struct dell_wmi_smbios_buffer *buffer, GError **error)
```

Comment:
In retrospect, I think I'd rather just depend upon a newer version of libsmbios_c that supports WMI interface natively rather than support this approach (that's in fwupdate today).  It's a lot less code that way at least.  I don't think it needs to happen in this PR (unlesss the stuff you did as is isn't yet functional), I'll do a follow up one with it.

Reply:
Could you modify this branch with what you'd like to remove? I'm happy depending on a new smbios release.